middleman-core/lib/middleman-core/preview_server.rb

```diff
@@ -110,24 +111,29 @@ def new_app
           run meta_app
         end
 
+        my_ip       = opts[:bind_address].presence || public_ip
+        my_hostname = opts[:host].presence || resolve_ip(my_ip) || Socket.gethostname.tr(' ', '+')
```

Comment:
Previously we had people complaining about having to do a reverse DNS lookup at any point (because they were on the train or whatever). What does `resolve_ip` get us that `Socket.gethostname` doesn't?